simulator/src/main/java/com/hazelcast/simulator/worker/WorkerPerformanceMonitor.java

```diff
@@ -41,6 +48,7 @@ boolean start() {
 
     void stop() {
         thread.stop = true;
+        thread.interrupt();
```

Comment:
Don't do this. interrupt is unreliable because you can interrupt the thread, but if that interrupt is gobbled up.. then the thread will never terminate. So always use an additional flag; interrupt is error prone for termination.